### PR TITLE
enable fedora 30 testing

### PR DESCRIPTION
**Pull request by rcerven**
and disable fedora 28 testing, build py2 packages only for
rhel <=7, and py3 for fedora and rhel >= 8

Signed-off-by: Robert Cerven <rcerven@redhat.com>

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request includes link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic-reactor.spec [hunk 1]
```diff
@@ -86,6 +69,23 @@ BuildRequires:  python3-six
...
+%else
+BuildRequires:  python2-devel
+BuildRequires:  python-setuptools
+%if 0%{?with_check}
+BuildRequires:  pytest
+BuildRequires:  python-dockerfile-parse >= 0.0.11
+%if (0%{?fedora} >= 29)
+BuildRequires:  python2-docker
+%else
+BuildRequires:  python-docker-py
```
Comment (athos-ribeiro) [inline atomic-reactor.spec]: This else block can probably go away: Fedora < 29 has reached EoL anyway (there is anothe occurrence of this case above which should also be removed).

File: atomic-reactor.spec [hunk 2]
```diff
@@ -86,6 +69,23 @@ BuildRequires:  python3-six
...
+%endif  # fedora 29+
+BuildRequires:  python-flexmock >= 0.10.2
+BuildRequires:  python-six
+BuildRequires:  python-osbs >= 0.48
+BuildRequires:  python-backports-lzma
+BuildRequires:  python2-responses
+%endif # with_check
 %endif # with_python3
```
Comment (athos-ribeiro) [inline atomic-reactor.spec]: Could we aggregate the with_check blocks in a single one?

File: atomic-reactor.spec (hunk 1 above)
Comment (rcerven) [reply atomic-reactor.spec]: ok

File: atomic-reactor.spec (hunk 2 above)
Comment (rcerven) [reply atomic-reactor.spec]: ok

File: atomic-reactor.spec [hunk 3]
```diff
@@ -98,6 +92,70 @@ probably implement if you started hooking Docker into your
 infrastructure.
 
 
+%if 0%{?with_python3}
+%package -n python3-atomic-reactor
+Summary:        Python 3 Atomic Reactor library
+Group:          Development/Tools
+License:        BSD
+%if (0%{?fedora} >= 29)
```
Comment (athos-ribeiro) [inline atomic-reactor.spec]: this block coud be removed as well

Comment (rcerven) [reply atomic-reactor.spec]: ok

**Comment by rcerven** [comment]
I'd be in favor of removing it, @twaugh  ?

**Comment by rcerven** [comment]
removed %check in spec file in 2nd commit

**Comment by rcerven** [comment]
rn updated: enable building py3 package for rhel8

**Pull request merged**